File: tests/com/github/redfoos/logstash/completion/CompletionTest.java
package com.github.redfoos.logstash.completion;

import com.intellij.testFramework.fixtures.BasePlatformTestCase;

public class CompletionTest extends BasePlatformTestCase {
    @Override
    protected String getTestDataPath() {
        return "testData/completion";
    }

    public void testI() {
        doTest("input", "if", "filter");
    }

    public void testE() {
        doTest("else", "filter");
    }

    private void doTest(String... completionVariants) {
        myFixture.testCompletionVariants(getTestFilePath(), completionVariants);
    }

    private String getTestFilePath() {
        return getTestName(true) + ".conf";
    }
}

File: src/test/kotlin/com/github/redfoos/logstash/completion/CompletionTest.kt
package com.github.redfoos.logstash.completion

import com.intellij.testFramework.fixtures.BasePlatformTestCase
import org.junit.Ignore
import org.junit.Test

class CompletionTest : BasePlatformTestCase() {
    override fun getTestDataPath(): String {
        return "src/test/testData/completion"
    }
    @Test
    fun testI() {
        doTest("input", "filter")
    }
    @Test
    fun testE() {
        doTest("filter")
    }

    @Test
    @Ignore
    fun testO() {
        doTest("output")
    }
    private fun doTest(vararg completionVariants: String) {
        myFixture.testCompletionVariants(getTestFilePath(), *completionVariants)
    }
    private fun getTestFilePath(): String {
        return "${getTestName(true)}.conf"
    }
}
